src/equilibrate_utils.jl

```diff
@@ -0,0 +1,726 @@
+    end
+    return best_solution
+end
+
+#Prepares the reaction network for being solved with Homotopy Continuation, in case some initialising operations have not been done yet.
+function initialise_solver!(rn::DiffEqBase.AbstractReactionNetwork, p::Vector{Float64}, bifurcation_exception_parameter::Integer=-1)
+    check_is_polynomial(rn)
+    using_temp_poly = !has_equi_poly(rn)
+    using_temp_poly && fix_parameters(rn, p, full_vector_exemption=bifurcation_exception_parameter)
+    !has_hc_templates(rn) && make_hc_template(rn)
+    return using_temp_poly
+end
+# In case a temporary equilibrium polynomial were used, this one resets it.
+function finalise_solver!(rn::DiffEqBase.AbstractReactionNetwork)
+    reset_equi_poly!(rn); reset_hc_templates!(rn);
+end
+# Extracts the biological plausible solutions (real and non negative).
+function positive_real_solutions(result)
+    tmp_sol = filter(result) do y
+        all(yᵢ -> abs(imag(yᵢ)) < 0.001, y)&&all(yᵢ -> real(yᵢ) ≥ -0.001, y)
+    end
+    return map(x->real.(x),tmp_sol)
+end
+
+#Returns a boolean which will be true if the given fixed point is stable.
+"""
+    stability(solution,reaction_network, parameter_values,  t=0.)
+
+    Gives the steady stability of the steady state of a certain network. Returns true if the point is stable, false if it is not.
+
+## args
+- solution: the values of the reactants in the steady state.
+- parameter_values: the parameter values in for which the steady state was calculated.
+- reaction_network: a reaction network.
+- t (optional): the value of t, for which the stability is calculated.
+"""
+function stability(solution::Vector{Float64}, rn::DiffEqBase.AbstractReactionNetwork, p::Vector{Float64}, t=0.::Float64)
+    jac = zeros(length(rn.syms),length(rn.syms))
+    rn.jac(jac,solution,p,t)
+    return maximum(real.(eigen(jac).values))<0.
+end
+#Runs stability on an vector of steady states.
+"""
+    stability(solutions,reaction_network, parameter_values,  t=0.)
+
+    Gives the steady stability of the steady state of a certain network.
+
+## args
+- solutions: An array of steady states (which each in turn, is an array of reactant concentrations in the steady state).
+- parameter_values: the parameter values in for which the steady state was calculated.
+- reaction_network: a reaction network.
+- t (optional): the value of t, for which the stability is calculated.
+"""
+function stability(solutions::Vector{Vector{Float64}}, rn::DiffEqBase.AbstractReactionNetwork, p::Vector{Float64}, t=0.::Float64)
+    return map(sol->stability(sol,rn,p,t), solutions)
+end
+
+
+
+### Structures holding information about bifurcation diagrams. ###
+
+### Bifurcation Diagrams Structures###
+
+#Contains information from a single bifurcation line. A bifurcation plot would typically contain several such lines. These are the results from a method which tracks a solution through parameter space.
+struct BifurcationPath
+    p_vals::Vector{Float64}
```

Comment:
Does it all need to be Float64s?

Reply:
Probably me being overzealous defining types more narrowly then they should. used `::Vector` would be better?

Reply:
By not restricting it at all you increase the users ability to draw on the rest of Julias ecosystem. You don't have to sacrifice performance if you do something like:

```julia
struct BifurcationPath{T} where T
    p_vals::T
    ...
```

However, just using `::Vector` would be bad for performance since the compiler cannot figure out what you are giving it.